annotations/upload_validation.py

```
@@ -0,0 +1,89 @@
+'''
+upload_validation.py
+author: aditi.gopalan
+Description: This script checks if manifests have been uploaded to Synapse 
+'''
+
+import synapseclient
+import pandas as pd
+import csv
+import argparse
+from datetime import datetime
+
+def get_folder_type_argument():
+    parser = argparse.ArgumentParser(description="Process Synapse folders and create a manifest.")
+    parser.add_argument("csv_file_path", help="Path to the input CSV file.")
+    parser.add_argument("folder_type", choices=["Publication", "Dataset", "Tool"],
+                        help="Specify folder type: 'Publication', 'Dataset', or 'Tool'")
+    args = parser.parse_args()
+    return args.csv_file_path, args.folder_type
+
+def get_folder_id_column(folder_type):
+    if folder_type == "Publication":
+        return "folderIdPublication"
+    elif folder_type == "Dataset":
+        return "folderIdDatasets"
+    elif folder_type == "Tool":
+        return "folderIdTools"
+
+def get_project_name(syn, folder_id):
+    try:
+        folder = syn.get(folder_id)
+        project = syn.get(folder.parentId)
+        return project.name
+    except Exception as e:
```

Nit: it's usually more ideal to be as specific as possible when catching exceptions rather than generally catch all exceptions.  That way, for any exceptions that do arise that are outside of expectations, let's say like an `IndexError` or `FileNotFoundError`, we can address them accordingly.

Of course, this is all situational!  I think for the purpose of this script, catching all exceptions is appropriate.  Just an FYI for future scripts 😁

---

Makes sense, these tips are super helpful thank you!!